metagpt/roles/role.py

```diff
@@ -374,7 +374,7 @@ async def _think(self) -> bool:
             if self.rc.max_react_loop != len(self.actions):
                 self.rc.max_react_loop = len(self.actions)
             self._set_state(self.rc.state + 1)
-            return self.rc.state >= 0 and self.rc.state < len(self.actions)
+            return (self.rc.state + 1) >= 0 and (self.rc.state + 1) < len(self.actions)
```

Comment:
`self._set_state` has incremented self.rc.state and therefore no need for another increment